### PR TITLE
Enhance network hostname documentation.

### PR DESCRIPTION
**Pull request by jukkar**

### CODE REVIEW

**Review by nashif — APPROVED** [review]

File: doc/reference/networking/net_hostname.rst [hunk 1]
```diff
@@ -6,6 +6,26 @@ Hostname Configuration
 Overview
 ********
 
+A networked device might need a hostname in certain cases like if the device
```
Comment (dbkinder) [inline doc/reference/networking/net_hostname.rst]:
```suggestion
A networked device might need a hostname, for example, if the device
```

File: doc/reference/networking/net_hostname.rst [hunk 2]
```diff
@@ -6,6 +6,26 @@ Hostname Configuration
...
+is configured to be a mDNS responder (see :ref:`dns_resolve_interface` for
+details) and needs to respond to ``<hostname>.local`` DNS queries.
+
+The :option:`CONFIG_NET_HOSTNAME_ENABLE` must be set in config file in order
```
Comment (dbkinder) [inline doc/reference/networking/net_hostname.rst]: set in which config file?  or should we just delete "in config file"

File: doc/reference/networking/net_hostname.rst [hunk 3]
```diff
@@ -6,6 +6,26 @@ Hostname Configuration
...
+:option:`CONFIG_NET_HOSTNAME` option.
+
+If the same firmware image is used to flash multiple boards, then it is not
+practical to use the same hostname in all of the boards. In that case, one
+can enable :option:`CONFIG_NET_HOSTNAME_UNIQUE` which will add a unique
+postfix to the hostname. By default link address of the first network interface
+is used as a postfix. In Ethernet networks, the link address refers to MAC
+address. For example, if the link address is ``01:02:03:04:05:06``, then the
+unique hostname could be ``zephyr010203040506``. If you want to set the prefix
+yourself, then make sure to call ``net_hostname_set_postfix()`` before the
```
Comment (dbkinder) [inline doc/reference/networking/net_hostname.rst]:
```suggestion
yourself, then call ``net_hostname_set_postfix()`` before the
```

**Review by dbkinder — DISMISSED** [review]

File: doc/reference/networking/net_hostname.rst (hunk 2 above)
Comment (jukkar) [reply doc/reference/networking/net_hostname.rst]: I will delete the "in config file" part, it is useless.

**Comment by jukkar** [comment]
Updated according to comments.

**Review by dbkinder — APPROVED** [review]
looks good, thanks!

**Pull request merged**